app/controllers/concerns/cookies.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Cookies
+  extend ActiveSupport::Concern
+
+  include ActionController::Cookies
+
+  def set_session_id_cookie(session)
+    return unless session in Session # lol nice
+    return unless
+      request.origin.nil? || request.origin.ends_with?(Keygen::DOMAIN)
+
+    cookies.encrypted[:session_id] = {
+      value: session.id,
+      expires: session.expiry,
+      domain: Keygen::DOMAIN,
+      same_site: :lax,
```

Comment:
I'm so confused about same-site cookies esp i.r.t. CSRF. Everything online is ambiguous at best, or wrong at worst.

Goal is to allow `fetch()` with cookies sent from `.keygen.sh` origins, but not from `evil.example`.